### PR TITLE
fix: typeError: Cannot read properties of undefined (reading 'split')

### LINKED ISSUE #3473: Crash report: `TypeError: Cannot read properties of undefined (reading 'split')`
## URL
  
https://app.uniswap.org/#/vote?chain=mainnet

## Error

```
TypeError: Cannot read properties of undefined (reading 'split')
```

## Stacktrace

```
TypeError: Cannot read properties of undefined (reading 'split')
    at https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:16005
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:15826
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:15266
    at Object._i [as useMemo] (https://app.uniswap.org/static/js/35.0620f59e.chunk.js:2:3432674)
    at t.useMemo (https://app.uniswap.org/static/js/35.0620f59e.chunk.js:2:3350521)
    at J (https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:14990)
    at ne (https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:16764)
    at kt (https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:50793)
```

## Device data

```json
{
  "ua": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36",
  "browser": {
    "name": "Chrome",
    "version": "99.0.4844.51",
    "major": "99"
  },
  "engine": {
    "name": "Blink",
    "version": "99.0.4844.51"
  },
  "os": {
    "name": "Mac OS",
    "version": "10.15.7"
  },
  "device": {},
  "cpu": {}
}
```

### LINKED ISSUE #3489: Crash report: `TypeError: Cannot read properties of undefined (reading 'split')`
## URL
  
https://app.uniswap.org/#/vote

## Error

```
TypeError: Cannot read properties of undefined (reading 'split')
```

## Stacktrace

```
TypeError: Cannot read properties of undefined (reading 'split')
    at https://app.uniswap.org/static/js/77.8031dd67.chunk.js:1:2903
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/77.8031dd67.chunk.js:1:2724
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/77.8031dd67.chunk.js:1:2166
    at Object.Aa [as useMemo] (https://app.uniswap.org/static/js/34.d906cff3.chunk.js:2:1254242)
    at t.useMemo (https://app.uniswap.org/static/js/34.d906cff3.chunk.js:2:1189261)
    at L (https://app.uniswap.org/static/js/77.8031dd67.chunk.js:1:1890)
    at K (https://app.uniswap.org/static/js/77.8031dd67.chunk.js:1:3649)
    at Ot (https://app.uniswap.org/static/js/77.8031dd67.chunk.js:1:37447)
```

## Device data

```json
{
  "ua": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36",
  "browser": {
    "name": "Chrome",
    "version": "99.0.4844.51",
    "major": "99"
  },
  "engine": {
    "name": "Blink",
    "version": "99.0.4844.51"
  },
  "os": {
    "name": "Windows",
    "version": "10"
  },
  "device": {},
  "cpu": {
    "architecture": "amd64"
  }
}
```

### LINKED ISSUE #3490: Crash report: `TypeError: Cannot read properties of undefined (reading 'split')`
## URL
  
https://app.uniswap.org/#/vote?chain=mainnet

## Error

```
TypeError: Cannot read properties of undefined (reading 'split')
```

## Stacktrace

```
TypeError: Cannot read properties of undefined (reading 'split')
    at https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:16005
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:15826
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:15266
    at Object._i [as useMemo] (https://app.uniswap.org/static/js/35.0620f59e.chunk.js:2:3432674)
    at t.useMemo (https://app.uniswap.org/static/js/35.0620f59e.chunk.js:2:3350521)
    at J (https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:14990)
    at ne (https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:16764)
    at kt (https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:50793)
```

## Device data

```json
{
  "ua": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36",
  "browser": {
    "name": "Chrome",
    "version": "99.0.4844.51",
    "major": "99"
  },
  "engine": {
    "name": "Blink",
    "version": "99.0.4844.51"
  },
  "os": {
    "name": "Windows",
    "version": "10"
  },
  "device": {},
  "cpu": {
    "architecture": "amd64"
  }
}
```

### LINKED ISSUE #3494: Crash report: `TypeError: Cannot read properties of undefined (reading 'split')`
## URL
  
https://app.uniswap.org/#/vote?chain=mainnet

## Error

```
TypeError: Cannot read properties of undefined (reading 'split')
```

## Stacktrace

```
TypeError: Cannot read properties of undefined (reading 'split')
    at https://app.uniswap.org/static/js/42.f1d87e3b.chunk.js:1:16005
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/42.f1d87e3b.chunk.js:1:15826
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/42.f1d87e3b.chunk.js:1:15266
    at Object._i [as useMemo] (https://app.uniswap.org/static/js/35.0620f59e.chunk.js:2:3432674)
    at t.useMemo (https://app.uniswap.org/static/js/35.0620f59e.chunk.js:2:3350521)
    at J (https://app.uniswap.org/static/js/42.f1d87e3b.chunk.js:1:14990)
    at ne (https://app.uniswap.org/static/js/42.f1d87e3b.chunk.js:1:16764)
    at Nt (https://app.uniswap.org/static/js/42.f1d87e3b.chunk.js:1:52095)
```

## Device data

```json
{
  "ua": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36",
  "browser": {
    "name": "Chrome",
    "version": "99.0.4844.51",
    "major": "99"
  },
  "engine": {
    "name": "Blink",
    "version": "99.0.4844.51"
  },
  "os": {
    "name": "Windows",
    "version": "10"
  },
  "device": {},
  "cpu": {
    "architecture": "amd64"
  }
}
```

### LINKED ISSUE #3501: Crash report: `TypeError: Cannot read properties of undefined (reading 'split')`
## URL
  
https://app.uniswap.org/#/vote?chain=mainnet

## Error

```
TypeError: Cannot read properties of undefined (reading 'split')
```

## Stacktrace

```
TypeError: Cannot read properties of undefined (reading 'split')
    at https://app.uniswap.org/static/js/79.a0bf15ae.chunk.js:1:2917
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/79.a0bf15ae.chunk.js:1:2738
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/79.a0bf15ae.chunk.js:1:2178
    at Object.Pa [as useMemo] (https://app.uniswap.org/static/js/35.5196e934.chunk.js:2:2958714)
    at t.useMemo (https://app.uniswap.org/static/js/35.5196e934.chunk.js:2:2876561)
    at z (https://app.uniswap.org/static/js/79.a0bf15ae.chunk.js:1:1902)
    at _ (https://app.uniswap.org/static/js/79.a0bf15ae.chunk.js:1:3663)
    at xt (https://app.uniswap.org/static/js/79.a0bf15ae.chunk.js:1:37720)
```

## Device data

```json
{
  "ua": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36",
  "browser": {
    "name": "Chrome",
    "version": "99.0.4844.51",
    "major": "99"
  },
  "engine": {
    "name": "Blink",
    "version": "99.0.4844.51"
  },
  "os": {
    "name": "Windows",
    "version": "10"
  },
  "device": {},
  "cpu": {
    "architecture": "amd64"
  }
}
```

### LINKED ISSUE #3506: Crash report: `TypeError: Cannot read properties of undefined (reading 'split')`
## URL
  
https://app.uniswap.org/#/vote?chain=mainnet

## Error

```
TypeError: Cannot read properties of undefined (reading 'split')
```

## Stacktrace

```
TypeError: Cannot read properties of undefined (reading 'split')
    at https://app.uniswap.org/static/js/79.7130818e.chunk.js:1:2917
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/79.7130818e.chunk.js:1:2738
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/79.7130818e.chunk.js:1:2178
    at Object.Pa [as useMemo] (https://app.uniswap.org/static/js/35.d7634141.chunk.js:2:2958714)
    at t.useMemo (https://app.uniswap.org/static/js/35.d7634141.chunk.js:2:2876561)
    at z (https://app.uniswap.org/static/js/79.7130818e.chunk.js:1:1902)
    at _ (https://app.uniswap.org/static/js/79.7130818e.chunk.js:1:3663)
    at xt (https://app.uniswap.org/static/js/79.7130818e.chunk.js:1:37720)
```

## Device data

```json
{
  "ua": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36",
  "browser": {
    "name": "Chrome",
    "version": "99.0.4844.51",
    "major": "99"
  },
  "engine": {
    "name": "Blink",
    "version": "99.0.4844.51"
  },
  "os": {
    "name": "Mac OS",
    "version": "10.15.7"
  },
  "device": {},
  "cpu": {}
}
```

### LINKED ISSUE #3521: Crash report: `TypeError: Cannot read properties of undefined (reading 'split')`
## URL
  
https://app.uniswap.org/#/vote?chain=mainnet

## Error

```
TypeError: Cannot read properties of undefined (reading 'split')
```

## Stacktrace

```
TypeError: Cannot read properties of undefined (reading 'split')
    at https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:16005
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:15826
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:15266
    at Object._i [as useMemo] (https://app.uniswap.org/static/js/35.0620f59e.chunk.js:2:3432674)
    at t.useMemo (https://app.uniswap.org/static/js/35.0620f59e.chunk.js:2:3350521)
    at J (https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:14990)
    at ne (https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:16764)
    at kt (https://app.uniswap.org/static/js/42.8ad82213.chunk.js:1:50793)
```

## Device data

```json
{
  "ua": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36",
  "browser": {
    "name": "Chrome",
    "version": "99.0.4844.51",
    "major": "99"
  },
  "engine": {
    "name": "Blink",
    "version": "99.0.4844.51"
  },
  "os": {
    "name": "Mac OS",
    "version": "10.15.7"
  },
  "device": {},
  "cpu": {}
}
```

### LINKED ISSUE #3522: Crash report: `TypeError: Cannot read properties of undefined (reading 'split')`
## URL
  
https://app.uniswap.org/#/vote

## Error

```
TypeError: Cannot read properties of undefined (reading 'split')
```

## Stacktrace

```
TypeError: Cannot read properties of undefined (reading 'split')
    at https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:2906
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:2727
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:2167
    at Object.Ci [as useMemo] (https://app.uniswap.org/static/js/34.8c7c00d7.chunk.js:2:2454178)
    at t.useMemo (https://app.uniswap.org/static/js/34.8c7c00d7.chunk.js:2:2389197)
    at q (https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:1891)
    at K (https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:3652)
    at Ot (https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:37747)
```

## Device data

```json
{
  "ua": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36",
  "browser": {
    "name": "Chrome",
    "version": "99.0.4844.51",
    "major": "99"
  },
  "engine": {
    "name": "Blink",
    "version": "99.0.4844.51"
  },
  "os": {
    "name": "Mac OS",
    "version": "10.15.7"
  },
  "device": {},
  "cpu": {}
}
```

### LINKED ISSUE #3533: Crash report: `TypeError: Cannot read properties of undefined (reading 'split')`
## URL
  
https://app.uniswap.org/#/vote

## Error

```
TypeError: Cannot read properties of undefined (reading 'split')
```

## Stacktrace

```
TypeError: Cannot read properties of undefined (reading 'split')
    at https://app.uniswap.org/static/js/78.adc2f8f0.chunk.js:1:2903
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/78.adc2f8f0.chunk.js:1:2724
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/78.adc2f8f0.chunk.js:1:2166
    at Object.Aa [as useMemo] (https://app.uniswap.org/static/js/35.5c687bd5.chunk.js:2:1364981)
    at t.useMemo (https://app.uniswap.org/static/js/35.5c687bd5.chunk.js:2:1300000)
    at L (https://app.uniswap.org/static/js/78.adc2f8f0.chunk.js:1:1890)
    at K (https://app.uniswap.org/static/js/78.adc2f8f0.chunk.js:1:3649)
    at Ot (https://app.uniswap.org/static/js/78.adc2f8f0.chunk.js:1:37448)
```

## Device data

```json
{
  "ua": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36 Edg/99.0.1150.36",
  "browser": {
    "name": "Edge",
    "version": "99.0.1150.36",
    "major": "99"
  },
  "engine": {
    "name": "Blink",
    "version": "99.0.4844.51"
  },
  "os": {
    "name": "Windows",
    "version": "10"
  },
  "device": {},
  "cpu": {
    "architecture": "amd64"
  }
}
```

### LINKED ISSUE #3534: Crash report: `TypeError: Cannot read properties of undefined (reading 'split')`
## URL
  
https://app.uniswap.org/#/vote

## Error

```
TypeError: Cannot read properties of undefined (reading 'split')
```

## Stacktrace

```
TypeError: Cannot read properties of undefined (reading 'split')
    at https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:2906
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:2727
    at Array.map (<anonymous>)
    at https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:2167
    at Object.Ci [as useMemo] (https://app.uniswap.org/static/js/34.8c7c00d7.chunk.js:2:2454178)
    at t.useMemo (https://app.uniswap.org/static/js/34.8c7c00d7.chunk.js:2:2389197)
    at q (https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:1891)
    at K (https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:3652)
    at Ot (https://app.uniswap.org/static/js/78.77d898da.chunk.js:1:37747)
```

## Device data

```json
{
  "ua": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.74 Safari/537.36",
  "browser": {
    "name": "Chrome",
    "version": "99.0.4844.74",
    "major": "99"
  },
  "engine": {
    "name": "Blink",
    "version": "99.0.4844.74"
  },
  "os": {
    "name": "Windows",
    "version": "10"
  },
  "device": {},
  "cpu": {
    "architecture": "amd64"
  }
}
```

### PR DESCRIPTION
**Pull request by zhyd1997**
fixes #3506 
fixes #3501 
fixes #3494 
fixes #3490 
fixes #3489 
fixes #3473 
fixes #3521 
fixes #3522 
fixes #3533 
fixes #3534

### CODE REVIEW

**Comment by zhyd1997** [comment]
@justindomingue 
@NoahZinsmeister 
@ianlapham 

Please take a look, thanks!

File: src/state/governance/hooks.ts [hunk 1]
```diff
@@ -269,7 +269,7 @@ export function useAllProposalData(): { data: ProposalData[]; loading: boolean }
       data: proposalsCallData.map((proposal, i) => {
         const startBlock = parseInt(proposal?.result?.startBlock?.toString())
 
-        let description = formattedLogs[i]?.description
+        let description = formattedLogs.length ? formattedLogs[i]?.description : ''
```
Comment (justindomingue) [inline src/state/governance/hooks.ts:272]:
```suggestion
        let description = formattedLogs[i]?.description ?? ''
```

is this equivalent?

**Comment by zhyd1997** [comment]
@justindomingue 
Could this PR be merged?

**Comment by justindomingue** [comment]
@zhyd1997 thank you! sorry was letting others some time to review if needed

**Pull request merged**